File: apiJava/demoApiProva/demoApiProva/src/main/java/com/example/demoApiProva/model/Animal.java
package com.example.demoApiProva.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "animal")
public class Animal {

    @Id
    private String nome;

    @Column
    private String filo;

    @Column
    private int tempoVida;

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getFilo() {
        return filo;
    }

    public void setFilo(String filo) {
        this.filo = filo;
    }

    public int getTempoVida() {
        return tempoVida;
    }

    public void setTempoVida(int tempoVida) {
        this.tempoVida = tempoVida;
    }
}

File: exerciciosFlutter/flutter_application_prova_api_segundo_semestre/lib/animal.dart
class Animal {

  String _nome = "";
  String _filo = "";
  int _tempoVida = 0;
 
 String get nome => this._nome;

 set nome(String value) => this._nome = value;

  get filo => this._filo;

 set filo( value) => this._filo = value;

  get tempoVida => this._tempoVida;

 set tempoVida( value) => this._tempoVida = value;

 Animal();

 Animal.fromJson(Map<String, dynamic> json):
 _nome = json['nome'],
 _filo = json['filo'],
 _tempoVida = json['tempoVida']
 ;


}